pkg/eventpersister/persister/persister.go

```diff
@@ -666,11 +667,17 @@ func (p *Persister) createUserEvent(
 func (p *Persister) upsertEvaluationCount(event proto.Message, environmentNamespace string) error {
 	if e, ok := event.(*eventproto.EvaluationEvent); ok {
 		eck := p.key(eventCountKey, e.FeatureId, e.VariationId, environmentNamespace, e.Timestamp)
+		if err := p.setExpiration(eck); err != nil {
```

Comment:
This implementation will query every time to check if the key exists, which is not efficient.
What we need is to check when the count is created.

E.g.
```go
count, err := p.evaluationCountCacher.Increment(eck)
if err != nil {
	return err
}
if count == 1 {
  p.evaluationCountCacher.Expire(key, time)
}
```

Doing this solves the efficiency problem, but if the EXPIRE fails or a race condition happens, the key could stay in the memory.

We need to execute the INCR and EXPIRE as one command to fix that.
You probably will need to use a Lua script to do this.
Reference:
https://stackoverflow.com/a/21897569

First, I would recommend checking the go-redis library to see if there is a way to achieve this efficiently.

Also, how are you planning to update the current keys in the memory without the expiration?